backend/src/controllers/tasklist.js

```diff
@@ -37,6 +56,42 @@ const createTasklist = async (req, res) => {
   }
 };
 
+/**
+ * Deletes a given Tasklist and all corresponding tasks
+ *
+ * @param {*} req The incoming API request
+ * @param {*} res The outgoing API response
+ * @returns The status code only if request was processed correctly, or the error and status if request failed
+ */
+const deleteTasklist = async (req, res) => {
+  try {
+    const results = validateSchema(req.body, deleteTasklistSchema); //checks to see if valid list
+
+    if (!results.errors.length) {
+      const id = new mongoose.Types.ObjectId(req.body.tasklistId);
+
+      if (
+        (await Tasklist.findOne({ owner: req.user.address }).lean()) == null
+      ) {
+        return res.status(400).send("tasklist not found");
+      }
+
+      await Tasklist.findByIdAndDelete({
+        _id: id,
```

Comment:
Did you make it also delete all the tasks inside the tasklist?

Reply:
are we doing that through the schema or are do you want me to do it through the delete endpoint?

Reply:
Which ever you prefer, as long as it does it